### ISSUE TITLE
http://www.fullmatchesandshows.com/

### ISSUE THREAD

**Issue by satkiba** [reek/anti-adblock-killer#2877]
http://www.fullmatchesandshows.com/

Videos now showing that adblock is detected.

You're using an ad blocker.
To view this video, please disable your ad blocker for our site.
Select your browser plugin to view instructions:
AdBlock	Adblock Plus 

Anyone knows how to bypass it?

**Comment by uBlock-user** [comment]
already patched force update your filters.

**Comment by jspenguin2017** [comment]
Everything updated but still block screen. 
Test URL: http://www.fullmatchesandshows.com/2017/02/21/serie-a-review-21st-february-2017/
I'll post my solution if I find one. 
Also, I will be coming here daily (if I have time) to solve 1 or 2 issues and add them to my Script and List.

**Comment by jspenguin2017** [comment]
`@@||aka-cdn-ns.adtechus.com/images/5205_adwords_300x250V1.jpg$image,domain=fullmatchesandshows.com`
This works for now... I bet it will change regularly...

This is the offending script: 
```JavaScript
  self.prototype.AdBlockTester = function() {
    /**
     * @return {undefined}
     */
    function init() {
      /** @type {Image} */
      var cubeImage = new Image;
      /** @type {function (): undefined} */
      cubeImage.onload = successCallback;
      /** @type {function (): undefined} */
      cubeImage.onerror = onError;
      /** @type {string} */
      cubeImage.src = "//aka-cdn-ns.adtechus.com/images/5205_adwords_300x250V1.jpg";
    }
    var t;
    /** @type {boolean} */
    var n = false;
    /** @type {Array} */
    var codeSegments = [];
    /** @type {Array} */
    var results = [];
    /**
     * @return {undefined}
     */
    var successCallback = function() {
      /** @type {boolean} */
      t = true;
      /** @type {number} */
      var i = 0;
      for (;i < codeSegments.length;i++) {
        var win = codeSegments[i];
        win();
      }
    };
    /**
     * @return {undefined}
     */
    var onError = function() {
      /** @type {boolean} */
      t = false;
      /** @type {number} */
      var i = 0;
      for (;i < results.length;i++) {
        var trace = results[i];
        trace();
      }
    };
    return{
      /**
       * @param {?} properties
       * @param {?} log
       * @return {undefined}
       */
      check : function(properties, log) {
        codeSegments.push(properties);
        results.push(log);
        if (!n) {
          /** @type {boolean} */
          n = true;
          init();
        }
      }
    };
  }();
```

**Comment by satkiba** [comment]
`@@||aka-cdn-ns.adtechus.com/images/5205_adwords_300x250V1.jpg$image,domain=fullmatchesandshows.com`

this is working for me for now. thanks a lot!

**Comment by uBlock-user** [comment]
The website itself is patched. However playwire has its own anti adblock check before loading videos.

It was first patched by `$elemhide`. So playwire folks put another check by checking the status load of this script -`https://s3.amazonaws.com/html5player.gamezone.com/zzzsss.js` last time when I replied.

Now it appears that they're following this issue tracker and to troll the user they changed the script again.

**Comment by jspenguin2017** [comment]
@uBlock-user Yea, they just changed the image, I'll put your rule into my List. 👍 
I also added domain restriction just in case. 
`||adtechus.com/images^$image,redirect=32x32-transparent.png,domain=fullmatchesandshows.com|playwire.com`

**Comment by jspenguin2017** [comment]
If they patch this again, we might need some setter getter hack to filter writing into `Zeus` variable. Patch `AdBlockTester` on read? Or just replace their player altogether. I don't see why they are following this issue tracker, they will end up breaking their website. I don't see how content script can fight userscript.

**Comment by uBlock-user** [comment]
We need to find a consistent one that can be patched and does't get involved in this game. The core issue is that I can take out the entire script, but as a consequence the player won't load either, otherwise this would have been patched long ago.

This is the script where they change the checks - http://cdn.playwire.com/bolt/js/zeus/zeus_global-ca28c8cb46-4c84a2a84f.js

**Comment by jspenguin2017** [comment]
@uBlock-user 
Is that file name static? It's never easy to patch the player, but we can patch `Zeus` instead. But this will require Userscript, can't be done in uBlock Origin.

**Comment by jspenguin2017** [comment]
@uBlock-user BTW I put together a JavaScript decompiler: https://x01x012013.github.io/JavaScriptDecompiler/

**Comment by uBlock-user** [comment]
Yes as far as I know, they won't bother changing it as they will only change the e.src value. Yeah I guess so, I don't mind it as I don't browse the websites itself and playwire on my end is blocked by default. I simply love taking this as a challenge.

**Comment by uBlock-user** [comment]
Is that decompiler  a javascript Deobfuscator like - http://jsbeautifier.org/ ? Cool.

**Comment by jspenguin2017** [comment]
@uBlock-user It has 5 tools together, and I chose JS Nice instead of JS Beautifier.

**Comment by uBlock-user** [comment]
I'm running your AdBlockProtector script in ViolentMonkey, works fine. You should mention that or add support for VM if needed.

**Comment by jspenguin2017** [comment]
@uBlock-user If it work fine, that's great, but unfortunately, I don't have time to test for multiple script host, so officially, it only supports Tampermonkey. 
Also I think I found the solution: 
```JavaScript
let val;
a.win.Object.defineProperty(a.win, "Zeus", {
    configurable: false,
    set: function(v) {
        val = v;
    },
    get: function() {
        a.out.log(val);
        try {
            val._adBlockDetected = false;
        } catch (err) {}
        return val;
    }
});
```

**Comment by uBlock-user** [comment]
Yeah it works just fine, it's light weight and doesn't hog resources on my browsers, so i might just keep it.

Will that snippet work ? Have you tested ?

**Comment by jspenguin2017** [comment]
I did a lot of optimization based on Reek's script. And it does come with the cost of compatibility, for example, Greasemonkey is not supported. I'm happy to see that it works with Violentmonkey, also, I'm glad that you liked it. 

For the snippet above, I tested it, and it worked. Obviously they can still patch it, but it would require some re-design of their script, and it will take them a LOT longer to patch their script each time, with a big risk of breaking their website for everyone. I added this to my Script, it should work for a month at least:

```JavaScript
if (a.domCmp(["fullmatchesandshows.com", "playwire.com"])) {
    //Modify property on read
    let val;
    a.win.Object.defineProperty(a.win, "Zeus", {
        configurable: false,
        set: function (v) {
            val = v;
        },
        get: function () {
            try {
                val._adBlockDetected = false;
            } catch (err) { }
            return val;
        }
    });
}
```

**Comment by uBlock-user** [comment]
I like your script since it doesn't run for all domains which kills performance as it did with reek's and doesn't hog resources comparitively and is compatible with uBO. Btw I have tested that code, works perfectly fine. Lets see how long till they patch this one.

**Comment by uBlock-user** [comment]
Btw `cinemablend.com` and `goalsarena.org` also use playwire players. Secondly, would you like to add disabling popups directly opened via window.open ? I found this one snippet through which we can kill all major window.open popups - https://github.com/gorhill/uBlock/issues/2383

**Comment by jspenguin2017** [comment]
@uBlock-user 
Unlike what you think, it does run on most domains (only 50-ish domains excluded), just like Reek's code. Reek's code kills performance because it uses RegExp to compare domain. RegExp can be 10 times slower than `indexOf`, where `endsWith` that I'm using is even faster than `indexOf`. Also my script does not do installation check (unless you are on my home page), which also help with performance.

**Comment by jspenguin2017** [comment]
@uBlock-user 
I think there are many websites that use playwire players, would it be a good idea to set it as a generic protector?

**Comment by uBlock-user** [comment]
From what I see, it's highly optimised in comparision to reek's code, which was a mess.

**Comment by jspenguin2017** [comment]
I don't think making it generic will break anything, from what I see using `var` won't throw an error, and locking `Zeus._adBlockDetected` shouldn't be a problem on a normal website. I'll put it into generic list, please let me know if it cause problems. 

Update: Added to 6.27 😄

**Comment by uBlock-user** [comment]
Alright then, do as you see fit.

**Comment by jspenguin2017** [comment]
@uBlock-user I see that you edited your comment before, don't worry about performance hit by this, there are MUCH more to optimize in the generic list. Defining a variable that won't be used by 99% of websites won't change anything.

**Comment by uBlock-user** [comment]
@X01X012013 Thanks, Would you add a snippet to disable popups/unders permanently ? I found this one here - #https://github.com/gorhill/uBlock/issues/2383

They're a menace and I have yet to find a permanent solution, most of them get killed through that one snippet and two or three of them are still unpatched which I'm hoping you can kill them similarly.

**Comment by jspenguin2017** [comment]
@uBlock-user Some websites will hide the real link inside `open()` so I can't just remove the function for all websites. We can use `a.filter("open");` to disable popup for specific domains.

**Comment by uBlock-user** [comment]
Ofcourse not, I'm not asking for a generic solution, but something like that needs to be added, popups/popunders are becoming increasingly bad with each day, so do what you feel best. For example, hanime.tv creates two popups, based on a cookie check, so if cookies are not found, they hit you with two poups and that stays like that for 24 hours only.

**Comment by jspenguin2017** [comment]
@uBlock-user 
Hum, yea, I see what you mean. But I would not be able to add it to uBO, and Userscript doesn't really have "custom rule" so complex anti-popup behaviors would have to go through me. I'm still testing new optimizations to make sure I didn't break anything, I will have a look in a moment.

**Comment by jspenguin2017** [comment]
@uBlock-user Well, that was easy, just one line: `a.readOnly("BetterJsPop", function () { });`

**Comment by uBlock-user** [comment]
this also works - `fullmatchesandshows.com##script:inject(abort-on-property-write.js, UAParser)`
